### DISCUSSION TITLE
How to get library paths without angular.json

### DISCUSSION THREAD

**Discussion by mehrad-rafigh** [nrwl/nx#15319] [General]
Hi everybody,

I used the `angular.json` file to get the paths of all projects in my workspace. But since the migration to 15.7 the `angular.json` has been deleted. 

Is there any `nx` command that could print the paths to my libraries?

**Comment by alexciesielski** [comment]
Yeah, all my generators broke after updating. How was this not a breaking change/a major version release?

**Reply by mehrad-rafigh** [reply]
Well the ppl at Nrwl didn't thought that people might rely on angular.json for custom generators. Its a very good question @alexciesielski

**Reply by AgentEnder** [reply]
Generators that use the @nrwl/devkit methods to read project configuration / get the list of projects with `getProjects(tree)` shouldn't be affected by the change. If you were manually reading from `angular.json` or `workspace.json`, you should update to use the devkit helpers.

**Reply by alexciesielski** [reply]
Thanks, I checked `getProjects(tree)` but it returns an empty object for me though.

![image](https://user-images.githubusercontent.com/17651898/226157355-4232608a-9671-4acd-bb02-b043938dfba5.png)

**Reply by AgentEnder** [reply]
That shouldn't be the case, do your projects show up properly in the graph

**Comment by AgentEnder** [comment]
@mehrad-rafigh you can pull the info you need from the project graph. Its possible that info will not be as easily available if you were using some form of shell script, but you could always write a thin JS wrapper and invoke it wiithin your shell script. As an example:

> print-graph.js
```javascript
const devkit = require('@nrwl/devkit')
(async () => {
  console.log(JSON.stringify(await createProjectGraphAsync()))
}}()
```

> my-shell-script.sh
```shell
node ./print-graph.js | jq....
```
